tools/generator/Java.Interop.Tools.Generator.Importers/XmlApiImporter.cs

```diff
@@ -110,9 +110,14 @@ public static Field CreateField (XElement elem)
 
 			if (elem.Attribute ("managedName") != null)
 				field.Name = elem.XGetAttribute ("managedName");
-			else
+			else {
 				field.Name = TypeNameUtilities.StudlyCase (char.IsLower (field.JavaName [0]) || field.JavaName.ToLower ().ToUpper () != field.JavaName ? field.JavaName : field.JavaName.ToLower ());
```

Comment:
Not part of the bug, obviously, but since I noticed it...

Shouldn't this use `.ToLowerInvariant()` and `.ToUpperInvariant()`, so that we avoid the ["Turkish-I" problem](https://haacked.com/archive/2012/07/05/turkish-i-problem-and-why-you-should-care.aspx/)?